### PR TITLE
feat(docker): support docker compose with hdfs

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: docker/docker-compose-with-hdfs.yml [hunk 1]
```diff
@@ -0,0 +1,305 @@
+---
+version: "3"
+services:
```
Comment (liurenjie1024) [inline docker/docker-compose-with-hdfs.yml:3]: Where is hadoop?

File: docker/docker-compose-with-hdfs.yml [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  compactor-0:
+    image: "ghcr.io/risingwavelabs/risingwave:RW_0801_HDFS_273-x86_64"
```
Comment (liurenjie1024) [inline docker/docker-compose-with-hdfs.yml:5]: Use a more formal version later?

File: docker/docker-compose-with-hdfs.yml (hunk 1 above)
Comment (wcy-fdu) [reply docker/docker-compose-with-hdfs.yml:3]: `hummock+hdfs`

File: docker/docker-compose-with-hdfs.yml (hunk 2 above)
Comment (wcy-fdu) [reply docker/docker-compose-with-hdfs.yml:5]: In the dockerfile for specific customers, `<the image you packed>` is used instead.

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Comment by wcy-fdu** [comment]
> You may want to add a doc for how to configure the hdfs path in the docker compose. I guess most people won't notice that they should modify this yaml.

Yes, there will be two doc, one is via docker, another one is via k8s. Both are ready, and I will add it after testing this PR(This PR mount all hadoop folder on the image).

**Pull request merged**